### ISSUE TITLE
Failed to add some YouTube videos

### ISSUE THREAD

**Issue by KillyMXI** [DominikPieper/obsidian-ReadItLater#204]
### Please check that this issue hasn't been reported before.

- [X] I searched previous [Bug Reports](https://github.com/DominikPieper/obsidian-ReadItLater/labels/type%3A%20bug) didn't find any similar reports.

### Expected Behavior

Note created for a YouTube video URL in my clipboard when I activate ReadItLater action.

### Current behaviour

Both on mobile and desktop I'm getting a popup with a message to check the console log.

Checking the console log on desktop, getting a stack trace per attempt:

```
plugin:obsidian-read-it-later:70 Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'map')
    at Si.parseApiResponse (plugin:obsidian-read-it-later:70:67318)
    at async Si.prepareNote (plugin:obsidian-read-it-later:70:66130)
    at async Jt.makeNote (plugin:obsidian-read-it-later:175:1257)
    at async Jt.createNote (plugin:obsidian-read-it-later:175:718)

plugin:obsidian-read-it-later:70 Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'map')
    at Si.parseApiResponse (plugin:obsidian-read-it-later:70:67318)
    at async Si.prepareNote (plugin:obsidian-read-it-later:70:66130)
    at async Jt.makeNote (plugin:obsidian-read-it-later:175:1257)
    at async Jt.createNote (plugin:obsidian-read-it-later:175:718)
parseApiResponse @ plugin:obsidian-read-it-later:70
await in parseApiResponse
processContent @ plugin:obsidian-read-it-later:175
eval @ plugin:obsidian-read-it-later:175
```

### Steps to reproduce

I identified these links that cause the issue:

- <https://youtu.be/UoAsDlUwjy0>
- <https://youtu.be/2BOyJqzJVC0>
- <https://youtu.be/uFsaw8ldJOI>

Any of these links in the clipboard causes the error when activating ReadItLater action.

I checked some previously added videos, and those links seem to still work - either asks what to do with the duplicate or created the note if I renamed the previous one.

### Which Operating Systems are you using?

- [X] Android
- [ ] iPhone/iPad
- [ ] Linux
- [ ] macOS
- [X] Windows

### Obsidian Version

1.7.7

### Plugin Version

0.10.0

### Checks

- [X] I have tried it with all other plugins disabled and the error still occurs

### Possible solution

Probably being more tolerant to missing data in API responses.

**Comment by adamluckdev** [comment]
Thanks for the report. Issue is fixed in new version [0.10.1](https://github.com/DominikPieper/obsidian-ReadItLater/releases/tag/0.10.1).

**Comment by KillyMXI** [comment]
Thank you for such a swift fix
